classes/announcement/DAO.php

```diff
@@ -116,14 +116,6 @@ public function getMany(Collector $query): LazyCollection
         });
     }
 
-    /**
-     * @copydoc EntityDAO::fromRow()
-     */
-    public function fromRow(object $row): Announcement
-    {
-        return parent::fromRow($row);
```

Comment:
This shouldn't be removed; it provides accurate type hinting for the return type.

Reply:
While I was adding type hints, I've added some harmless "generics" to the code:
![image](https://github.com/pkp/pkp-lib/assets/361921/46d6cc25-063e-459a-9988-6469dca100bf)

![image](https://github.com/pkp/pkp-lib/assets/361921/d8f522da-9a95-461c-8ec2-084b9294f671)

So the proper type hint should be covered by this:
![image](https://github.com/pkp/pkp-lib/assets/361921/825c4f11-fa8c-4a30-8276-536a146748c4)

Reply:
If it makes the IDEs used by the dev team happy, then I'm all for removing the subclass functions. But this is great stuff to change at the start of a dev cycle, and to bring to the team as a suggested change in practice so we're all aware of it. Any changes going in now should be limited to bug fixes until we get `stable-3_4_0` branched from `main`.